shell/display.h

```diff
@@ -177,33 +166,24 @@ class Display {
 
   std::shared_ptr<TextInput> m_text_input{};
 
-  struct info {
-    struct {
-      int32_t x;
-      int32_t y;
-      int32_t physical_width;
-      int32_t physical_height;
-      int32_t size;
-      int32_t subpixel;
-      std::string make;
-      std::string model;
-      int32_t transform;
-    } geometry;
-
-    struct {
-      int32_t width;
-      int32_t height;
-      double dots_per_in;
-    } mode{};
-
-    struct {
-      int32_t factor;
-    } scale{};
-
-    std::string name;
-    std::string description;
-
-  } m_info;
+  typedef struct output_info {
+    struct wl_output* output;
+    uint32_t global_id;
+    unsigned width;
+    unsigned height;
+    unsigned physical_width;
+    unsigned physical_height;
+    int refresh_rate;
+    int32_t scale;
+    bool done;
+  } output_info_t;
```

Comment:
So... .wouldn't this better be a class? Instead of typedefing stuff. Its CPP after all.

Reply:
If you search the source tree you'll find many references to this pattern.  The goal is to stay as close to Google/Flutter standards.  This makes upstreaming much easier.  So this is fine.